brave-grpc/src/test/java/com/github/kristofa/brave/grpc/BraveGrpcInterceptorsTest.java

```diff
@@ -0,0 +1,173 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class BraveGrpcInterceptorsTest {
+
+    private static final HelloRequest HELLO_REQUEST = HelloRequest.newBuilder()
+        .setName("brave")
+        .build();
+
+    private Server server;
+    private ManagedChannel channel;
+    private Brave brave;
+    private boolean enableSampling;
+
+    @Before
+    public void before() throws Exception {
+        enableSampling = true;
+        SpanCollectorForTesting.getInstance().getCollectedSpans().clear();
+
+        final Brave.Builder builder = new Brave.Builder();
+        brave = builder
+            .spanCollector(SpanCollectorForTesting.getInstance())
+            .traceSampler(new ExplicitSampler())
+            .build();
+
+        int serverPort = pickUnusedPort();
+        server = ServerBuilder.forPort(serverPort)
+            .addService(ServerInterceptors.intercept(
+                GreeterGrpc.bindService(new GreeterImpl()), new BraveGrpcServerInterceptor(brave)))
+            .build()
+            .start();
+
+        channel = ManagedChannelBuilder.forAddress("localhost", serverPort)
+            .intercept(new BraveGrpcClientInterceptor("helloWorldCaller", brave))
+            .usePlaintext(true)
+            .build();
+    }
+
+    @Test
+    public void testBlockingUnaryCall() throws Exception {
+        GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+        HelloReply reply = stub.sayHello(HELLO_REQUEST);
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+    }
+
+    @Test
+    public void testAsyncUnaryCall() throws Exception {
+        GreeterFutureStub futureStub = GreeterGrpc.newFutureStub(channel);
+        ListenableFuture<HelloReply> helloReplyListenableFuture = futureStub.sayHello(HELLO_REQUEST);
+        HelloReply reply = helloReplyListenableFuture.get();
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+    }
+
+    @Test
+    public void usesExistingTraceId() throws Exception {
+        LocalTracer localTracer = brave.localTracer();
+        SpanId spanId = localTracer.startNewSpan("localSpan", "myop");
+        GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+        HelloReply reply = stub.sayHello(HELLO_REQUEST);
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+        List<Span> spans = SpanCollectorForTesting.getInstance().getCollectedSpans();
+        Optional<Span> maybeSpan = spans.stream()
+            .filter(s -> s.getAnnotations().stream().anyMatch(a -> "ss".equals(a.value)))
+            .findFirst();
+        assertTrue("Could not find expected server span", maybeSpan.isPresent());
+        Span span = maybeSpan.get();
+        assertThat(span.getTrace_id(), is(equalTo(spanId.getTraceId())));
+        assertThat(span.getParent_id(), is(equalTo(spanId.getSpanId())));
+    }
+
+    @Test
```

Comment:
I'd add a javadoc clarifying what this means.

If sampling status isn't propagated, the server would start a trace even if a client's request wasn't sampled. This tests that sampling status propagates and is honored by the server: both client and server spans are not collected.